### PR TITLE
refactor(VGrid): convert to TypeScript

### PR DESCRIPTION
**Pull request by sh7dm**
## Description
Convert `VGrid` to TS.

## Motivation and Context
TS conversion.

## How Has This Been Tested?
Markup + `jest`

## Markup:
<details>

```vue
<template>
  <v-app id="inspire">
    <v-navigation-drawer
      v-model="drawer"
      fixed
      app
    >
      <v-list dense>
        <v-list-tile @click="">
          <v-list-tile-action>
            <v-icon>home</v-icon>
          </v-list-tile-action>
          <v-list-tile-content>
            <v-list-tile-title>Home</v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
        <v-list-tile @click="">
          <v-list-tile-action>
            <v-icon>contact_mail</v-icon>
          </v-list-tile-action>
          <v-list-tile-content>
            <v-list-tile-title>Contact</v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
    </v-navigation-drawer>
    <v-toolbar color="indigo" dark fixed app>
      <v-toolbar-side-icon @click.stop="drawer = !drawer"></v-toolbar-side-icon>
      <v-toolbar-title>Application</v-toolbar-title>
    </v-toolbar>
    <v-content>
      <v-container grid-list-md text-xs-center>
        <v-layout row wrap>
          <v-flex xs12>
            <v-card dark color="primary">
              <v-card-text class="px-0">12</v-card-text>
            </v-card>
          </v-flex>
          <v-flex v-for="i in 2" :key="`6${i}`" xs6>
            <v-card dark color="secondary">
              <v-card-text class="px-0">6</v-card-text>
            </v-card>
          </v-flex>
          <v-flex v-for="i in 3" :key="`4${i}`" xs4>
            <v-card dark color="primary">
              <v-card-text class="px-0">4</v-card-text>
            </v-card>
          </v-flex>
          <v-flex v-for="i in 4" :key="`3${i}`" xs3>
            <v-card dark color="secondary">
              <v-card-text class="px-0">3</v-card-text>
            </v-card>
          </v-flex>
          <v-flex v-for="i in 6" :key="`2${i}`" xs2>
            <v-card dark color="primary">
              <v-card-text class="px-0">2</v-card-text>
            </v-card>
          </v-flex>
          <v-flex v-for="i in 12" :key="`1${i}`" xs1>
            <v-card dark color="secondary">
              <v-card-text class="px-0">1</v-card-text>
            </v-card>
          </v-flex>
        </v-layout>
      </v-container>
    </v-content>
    <v-footer color="indigo" app>
      <span class="white--text">&copy; 2017</span>
    </v-footer>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
      drawer: null
    }),
    props: {
      source: String
    }
  }
</script>
```
</details>

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VGrid/VContent.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 // Mixins
 import SSRBootable from '../../mixins/ssr-bootable'
 
 /* @vue/component */
-export default {
+export default Vue.extend({
   name: 'v-content',
 
-  mixins: [SSRBootable],
+  mixins: [ SSRBootable ],
```
[the remark this answers is not in the document]
Comment (sh7dm) [reply packages/vuetify/src/components/VGrid/VContent.ts]: @KaelWD OK, I've fixed that!

File: packages/vuetify/src/components/VGrid/grid.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
-export default function Grid (name) {
+import Vue, { VNode } from 'vue'
+
+export default function Grid (name: string) {
   /* @vue/component */
```
Comment (KaelWD) [inline packages/vuetify/src/components/VGrid/grid.ts]: remove this comment

File: packages/vuetify/src/components/VGrid/grid.ts [hunk 3]
```diff
@@ -13,7 +15,7 @@ export default function Grid (name) {
       }
     },
 
-    render: (h, { props, data, children }) => {
+    render: (h, { props, data, children }): VNode => {
```
Comment (KaelWD) [inline packages/vuetify/src/components/VGrid/grid.ts]: shouldn't be an arrow function

File: packages/vuetify/src/components/VGrid/grid.ts (hunk 2 above)
Comment (sh7dm) [reply packages/vuetify/src/components/VGrid/grid.ts]: @KaelWD `/* @vue/component */` one?

File: packages/vuetify/src/components/VGrid/grid.ts [hunk 4]
```diff
@@ -22,6 +24,7 @@ export default function Grid (name) {
           // https://github.com/vuejs/vue/issues/7841
           if (key === 'slot') return false
 
+          data.attrs = data.attrs || []
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VGrid/grid.ts]:
```suggestion
          data.attrs = data.attrs || {}
```

**Comment by johnleider** [comment]
Closing due to inactivity. Also, please create the PR as a branch on the repo.

**Pull request closed without merging**